src/utils/git/diff.go

```diff
@@ -1,30 +1,27 @@
 package utils
 
 import (
-	"os/exec"
+	"github.com/aja-video/contra/src/configuration"
+	"github.com/pmezard/go-difflib/difflib"
+	"io/ioutil"
+	"log"
 )
 
-// GitDiff returns a diff.
-func GitDiff(path, filename string) (string, error) {
-	// Explicitly reminding ourselves that we're using exec.
-	return gitDiffExec(path, filename)
-}
-
-// gitDiffExec uses os/exec to pull a git diff.
-func gitDiffExec(path, filename string) (string, error) {
-	// Prep the command.
-	cmd := exec.Command("git", "diff", "-U4", "HEAD", filename)
-	// Switch to the workspace folder.
-	cmd.Dir = path
-	// Run!
-	diff, err := cmd.Output()
+// GitDiff returns a diff of the existing config and the collector output
+func GitDiff(config configuration.Config, filename string, output string) (string, error) {
```

Comment:
As mentioned above, perhaps this could become `func GitDiff(workspace string,` instead of passing in the configuration.